src/tag.rs

```diff
@@ -3,13 +3,25 @@ use crate::parser::{ParseError, ParseErrorKind, ParseResult};
 use crate::writer::{WriteBuf, WriteResult};
 
 #[derive(Debug, Copy, Clone, PartialEq, Eq, Hash)]
-pub(crate) enum TagClass {
+pub enum TagClass {
     Universal = 0b00,
     Application = 0b01,
     ContextSpecific = 0b10,
     Private = 0b11,
```

Comment:
As written, this makes the `u8` values into a public API, I'm not sure we want that. Can we make this a "regular" enum rather than one with fixed values?

Reply:
I used them, as the Implicit/Explicit do only support primitve const generic types. I now have a const fn method, that returns an u8 on a regular enum. 

I do not have a better Idea right now - if you have, just let me know.